core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -742,6 +742,25 @@ object EitherT extends EitherTInstances {
   final def liftK[F[_], A](implicit F: Functor[F]): F ~> EitherT[F, A, *] =
     new (F ~> EitherT[F, A, *]) { def apply[B](fb: F[B]): EitherT[F, A, B] = right(fb) }
 
+  /**
+   * Lifts an effect into EitherT, catching all errors from the effect and lifting them into EitherT's error channel.
+   *
+   * {{{
+   * scala> import cats._, data._, implicits._
+   * scala> val a: Option[Int] = None
+   * scala> val b: EitherT[Option, Unit, Int] = EitherT.liftAttemptK[Option, Unit].apply(a)
+   * scala> b.value
+   * res0: Option[Either[Unit, Int]] = Some(Left(()))
+   *
+   * scala> val a2: Option[Int] = Some(42)
+   * scala> val b2: EitherT[Option, Unit, Int] = EitherT.liftAttemptK[Option, Unit].apply(a2)
+   * scala> b2.value
+   * res1: Option[Either[Unit, Int]] = Some(Right(42))
+   * }}}
+   */
+  final def liftAttemptK[F[_], E](implicit F: ApplicativeError[F, E]): F ~> EitherT[F, E, *] =
+    λ[F ~> EitherT[F, E, *]](fa => EitherT(F.attempt(fa)))
```

Comment:
We've been removing the value-level kind-projector syntax for polymorphic functions, since there's no real possibility of it ever cross-building on Dotty, and since writing this out explicitly as:

```scala
new (F ~> EitherT[F, E, *]) { def apply[A](fa: F[A]): EitherT[F, E, A] = EitherT(F.attempt(fa)) }
```
…is only marginally less readable. I don't think that needs to be changed in this PR, though, since we're not actually cross-building on Dotty yet.

(Note that the type-level `λ` is fine on Dotty with `-Ykind-projector`.)

Reply:
oh, I was under the impression this was also part of the `-Ykind-projector` feature. But it makes sense, I'll keep it in mind.

Reply:
@kubukoz No, I hate kind-projector's value-level lambda syntax and couldn't be bothered to implement it in `-Ykind-projector`. :smile: (Also it would have required much more disruptive changes than anything else that made it into `-Ykind-projector`.)